src/session/client-certificate.c

```diff
@@ -41,6 +43,19 @@
 
 #define CLIENT_CERTIFICATE_DIRECTORY   "/run/cockpit/tls/clients"
 
+static int
+open_proc_pid (pid_t pid)
+{
+    char path[100];
+    int r = snprintf (path, sizeof (path), "/proc/%d", pid);
```

Comment:
I've seen some advice around that it's better to cast to a wider type than `int` for fields with unknown types like `pid_t`.  At the same time, I can't imagine a situation where pids would get bigger than `int`, but still might be worth it to be extra-paranoid here?  I've even seen a few people using `intmax_t` but that seems truly excessive.

Reply:
I cast/print it to a `long unsigned`. That should be ok? Nobody will ever need more than 2^64 pids!!1! :wink: